### ISSUE TITLE
modal and drawer responsive

### ISSUE THREAD

**Issue by mahmoodkd** [rsuite/rsuite#2156]
### What version of rsuite are you using?

4.9.4

### What version of React are you using?

17.0.0

### What browser are you using?

chrome,firefox

### Describe the Bug

Hi
Modal and drawer sizes are larger than the home screen in responsive mode. You can only select XS mode for that mode, which is still sometimes larger than the screen. Please make sure that it does not get larger than the home screen with any size selected.
Thanks for the good rsuite team

**Comment by simonguo** [comment]
Hi @mahmoodkd  You can set `full` prop on Drawer. It will be rendered as a percentage.
```
<Drawer full />
```

output: 
```css
.rs-drawer-full.rs-drawer-left {
    width: calc(100% - 60px);
}
```

![image](https://user-images.githubusercontent.com/1203827/143408487-c367ce90-804f-44e7-8e28-26bf7500f437.png)

@SevenOutman @LeightonYao Can we consider setting a max width?

**Comment by mahmoodkd** [comment]
yes "full" is good but I want when screen change width my modal or drawer only use "md" size

if I use full when screen width larger than mobile size my modal and drawer is fullscreen

**Comment by SevenOutman** [comment]
Hi @mahmoodkd, unfortunately rsuite is not designed to be responsive at the first place. However, I think it's good to implement responsiveness but I don't think it will become a high-priority task in our plans.

**Comment by mahmoodkd** [comment]
> 

Hi @SevenOutman thank you for your explanation, but rsuite is very strong and good, and if you consider this issue, it will be much better.

Thank you and good rsuite team.
